src/components/AddXpubDialog/AddXpubDialog.tsx

```diff
@@ -0,0 +1,102 @@
+import { CirclePlus } from 'lucide-react';
+import { Button } from '@/components/ui/button.tsx';
+import {
+  Dialog,
+  DialogContent,
+  DialogDescription,
+  DialogFooter,
+  DialogHeader,
+  DialogTitle,
+  DialogTrigger,
+} from '@/components/ui/dialog.tsx';
+import { Label } from '@/components/ui/label.tsx';
+import { Input } from '@/components/ui/input.tsx';
+import React, { useEffect, useState } from 'react';
+import { HD } from '@bsv/sdk';
+import { errorWrapper } from '@/utils';
+import { toast } from 'sonner';
+import { useDebounce } from 'use-debounce';
+import { useMutation, useQueryClient } from '@tanstack/react-query';
+import { useSpvWalletClient } from '@/contexts';
+
+export const AddXpubDialog = () => {
+  const [xPriv, setXPriv] = useState<string>('');
+  const [xPub, setXPub] = useState<string>('');
+  const [debouncedXPriv] = useDebounce(xPriv, 500);
```

Comment:
The `500` might be a global contact used also for all other debounces.

Reply:
I will create a global variable if be needed